src/main/java/org/spongepowered/common/data/datasync/entity/EntityCustomNameConverter.java

```diff
@@ -33,38 +33,35 @@
 import org.spongepowered.common.accessor.world.entity.EntityAccessor;
 import org.spongepowered.common.adventure.SpongeAdventure;
 import org.spongepowered.common.data.datasync.DataParameterConverter;
+
 import java.util.List;
 import java.util.Optional;
 
-public final class EntityCustomNameConverter extends DataParameterConverter<String> {
-
+public final class EntityCustomNameConverter extends DataParameterConverter<net.minecraft.network.chat.Component> {
     public EntityCustomNameConverter() {
         super(EntityAccessor.accessor$DATA_CUSTOM_NAME());
     }
 
     @Override
-    public Optional<DataTransactionResult> createTransaction(final Entity entity, final String currentValue, final String value) {
-        final Component currentText = SpongeAdventure.legacySection(currentValue);
-        final Component newValue = SpongeAdventure.legacySection(value);
-
+    public Optional<DataTransactionResult> createTransaction(final Entity entity, final net.minecraft.network.chat.Component oldValue, final net.minecraft.network.chat.Component newValue) {
         return Optional.of(DataTransactionResult.builder()
-                .replace(Value.immutableOf(Keys.DISPLAY_NAME, currentText))
-                .success(Value.immutableOf(Keys.DISPLAY_NAME, newValue))
+                .replace(Value.immutableOf(Keys.DISPLAY_NAME, SpongeAdventure.asAdventure(oldValue)))
```

Comment:
Should all of these be `CUSTOM_NAME` instead? `DISPLAY_NAME` is immutable so will never change, right?